llvm/lib/Transforms/Utils/FunctionComparator.cpp

```diff
@@ -207,22 +212,51 @@ int FunctionComparator::cmpMetadata(const Metadata *L,
 
   auto *CL = dyn_cast<ConstantAsMetadata>(L);
   auto *CR = dyn_cast<ConstantAsMetadata>(R);
-  if (CL == CR)
-    return 0;
-  if (!CL)
+  if (CL && CR) {
```

Comment:
Was this supposed to be?
```suggestion
  if (CL || CR) {
```